src/button-group/internal.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useImperativeHandle, useRef, forwardRef } from 'react';
+import { getBaseProps } from '../internal/base-component';
+import { ButtonGroupProps, InternalButtonGroupProps } from './interfaces';
+import { ButtonProps } from '../button/interfaces';
+import SpaceBetween from '../space-between/internal';
+import ItemElement from './item-element';
+import styles from './styles.css.js';
+import clsx from 'clsx';
+
+const InternalButtonGroup = forwardRef(
+  (
+    {
+      variant,
+      items = [],
+      onItemClick,
+      __internalRootRef = null,
+      dropdownExpandToViewport,
+      ...props
+    }: InternalButtonGroupProps,
+    ref: React.Ref<ButtonGroupProps.Ref>
+  ) => {
+    if (!variant) {
+      throw new Error('Invariant violation: ButtonGroup variant is not set.');
```

Comment:
I am thinking that maybe it is better to return `null` here and add a dev warnings (with `warnOnce` function) here. This will limit the blast radius in case the variant property will be missed on some accident in prod version. The fallback needs to be covered with unit tests.